src/java/org/apache/cassandra/db/compaction/unified/Controller.java

```diff
@@ -697,15 +817,49 @@ public static Controller fromOptions(CompactionRealm realm, Map<String, String>
                 baseShardCount = DEFAULT_BASE_SHARD_COUNT;
         }
 
-        double targetSStableSize = options.containsKey(TARGET_SSTABLE_SIZE_OPTION)
-                                   ? FBUtilities.parseHumanReadable(options.get(TARGET_SSTABLE_SIZE_OPTION), null, "B")
-                                   : DEFAULT_TARGET_SSTABLE_SIZE;
+        long targetSStableSize = options.containsKey(TARGET_SSTABLE_SIZE_OPTION)
+                                 ? FBUtilities.parseHumanReadableBytes(options.get(TARGET_SSTABLE_SIZE_OPTION))
+                                 : DEFAULT_TARGET_SSTABLE_SIZE;
+        long minSSTableSize = getSizeWithAltAndSpecial(options, MIN_SSTABLE_SIZE_OPTION, MIN_SSTABLE_SIZE_OPTION_MB, 20, MIN_SSTABLE_SIZE_OPTION_AUTO, MIN_SSTABLE_SIZE_AUTO, DEFAULT_MIN_SSTABLE_SIZE);
+        if (minSSTableSize > targetSStableSize * INVERSE_SQRT_2)
+            throw new ConfigurationException(String.format("The minimum sstable size %s cannot be larger than the target size's lower bound %s.",
+                                                           FBUtilities.prettyPrintMemory(minSSTableSize),
+                                                           FBUtilities.prettyPrintMemory((long) (targetSStableSize * INVERSE_SQRT_2))));
+
+        double sstableGrowthModifier = DEFAULT_SSTABLE_GROWTH;
```

Comment:
Would it be beneficial for the default `sstableGrowthModifier` to be determined by the `dataSetSize`? (i.e. if the `dataSetSize` is larger than some size then the default `sstableGrowthModifier` would increase from 0 to 0.25)